### ISSUE TITLE
Enhancement: Remove 2 year old backwards compatibility check from getParserServices

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5508]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

The `getParserServices` function contains this comment:

```ts
// backwards compatibility check
// old versions of the parser would not return any parserServices unless parserOptions.project was set
```

Additionally, later on in that file:

```ts
const hasFullTypeInformation =
  context.parserServices.hasFullTypeInformation ??
  /* backwards compatible */ true;
```

Does that mean we can get rid of some of the code in that file?

### Fail

```typescript
n/a
```

### Pass

```typescript
n/a
```

**Comment by JoshuaKGoldberg** [comment]
Not time sensitive at all, but assigning @bradzacher per the Git blame for triage.

**Comment by bradzacher** [comment]
This has been merged into the v6 branch
